nlp4all/models/DataSource.py

```diff
@@ -4,15 +4,26 @@
 This will be use to interface with individual users' data sources
 """
 
+from __future__ import annotations
+
 from sqlalchemy import Column, Integer, String, ForeignKey
+from sqlalchemy.orm import relationship, Mapped, mapped_column
 
-from .database import Base
+from .database import Base, NestedMutableJSONB, project_data_source_table
+from . import Data, Project
 
 class DataSource(Base): # pylint: disable=too-few-public-methods
     """DataSource class manages users' data sources"""
 
     __tablename__ = 'data_source'
 
     id = Column(Integer, primary_key=True)
-    user = Column(Integer, ForeignKey("user.id"), nullable=False)
-    data_source_name = Column(String(80), nullable=False)
+    projects: Mapped[list[Project]] = relationship(
+        secondary=project_data_source_table,
+        back_populates="data_sources")
+    structure: Mapped[dict] = mapped_column(NestedMutableJSONB, nullable=False)
```

Comment:
maybe add metadata and use that to retrieve path / json location of "main text", etc

Reply:
done